saleor/checkout/models.py

```diff
@@ -273,6 +275,20 @@ def get_country(self):
             self.set_country(country_code, commit=True)
         return country_code
 
+    def save_if_not_deleted(self, fields_to_update: Optional[list] = None) -> None:
```

Comment:
@korycins I know that we do not solve the problem that was mentioned on slack discussion, however this might solve the `DatabaseError` which is mainly caused by concurrency issue when checkout was removed in other transaction. Not sure if solving the general checkout issue should be resolved by this ticket